### PR TITLE
Update holds

### PR DESCRIPTION
**Pull request by johnwatson484**
https://eaflood.atlassian.net/browse/SFI-1658﻿

### CODE REVIEW

File: app/routes/home.js [hunk 1]
```diff
@@ -2,8 +2,11 @@ module.exports = {
   method: 'GET',
   path: '/',
   options: {
-    handler: (request, h) => {
-      return h.view('home')
+    handler: async (request, h) => {
+      return h.view('home', {
+        totalHolds: 0,
```
Comment (steve-dickinson) [inline app/routes/home.js:7]: Should this be calculating the actual counts for both holds and schemes?

Currently there are 2 schemes, but the cars/bucket is showing zero in the UI?

File: app/views/home.njk [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
+          <ul class="govuk-list">
+            <li>
+              <a href="/payment-holds" class="govuk-link">Manage holds</a>
+            </li>
+          </ul>
+        </div>
+        <div class="govuk-cards--2__item govuk-card--dashboard">
+          <h2 class="govuk-heading-m govuk-!-margin-bottom-2">Schemes</h2>
+          <p class="govuk-heading-xl govuk-!-margin-bottom-5">
+            {{ totalSchemes }}
```
Comment (steve-dickinson) [inline app/views/home.njk:23]: Total schemes is showing zero when there are 2 schemes.

Comment (steve-dickinson) [inline app/views/home.njk:23]: Total schemes is showing zero when there are 2 schemes.

File: app/views/payment-schemes.njk [hunk 3]
```diff
@@ -15,44 +15,37 @@
...
-                <td class="govuk-table__cell">{{ paymentScheme.active }}</td>
+                <td class="govuk-table__cell">{% if paymentScheme.active %}Active{% else %}Inactive{% endif %}</td>
                 <td class="govuk-table__cell">
                   <form method="POST" action="/payment-schemes">
                     <input type="hidden" name="crumb" value="{{crumb}}"/>
                     <input type="hidden" id="schemeId" name="schemeId" value="{{ paymentScheme.schemeId }}">
                     <input type="hidden" id="name" name="name" value="{{ paymentScheme.name }}">
                     <input type="hidden" id="active" name="active" value="{{ paymentScheme.active }}">
-                    <button class="govuk-button" data-module="govuk-button">Update</button>
+                    <button class="govuk-button govuk-button--secondary" data-module="govuk-button" style="margin-bottom: 0px;">{% if paymentScheme.active %}Disable{% else %}Enable{% endif %}</button>
```
Comment (steve-dickinson) [inline app/views/payment-schemes.njk:40]: Is a secondary button the correct button? Looks as though the button is disabled?

Comment (steve-dickinson) [inline app/views/payment-schemes.njk:40]: Is a secondary button the correct button? Looks as though the button is disabled?

File: app/views/payment-holds.njk [hunk 4]
```diff
@@ -41,13 +45,19 @@
                 <td class="govuk-table__cell">{{ paymentHold.holdCategoryName }}</td>
                 <td class="govuk-table__cell">{{ paymentHold.holdCategorySchemeName }}</td>
                 <td class="govuk-table__cell">{{ paymentHold.dateTimeAdded }}</td>
-                <td class="govuk-table__cell">{{ paymentHold.dateTimeClosed }}</td>
+                <td class="govuk-table__cell">
+                  <form method="POST" action="/remove-payment-hold">
+                    <input type="hidden" name="crumb" value="{{crumb}}"/>
+                    <input type="hidden" id="holdId" name="holdId" value="{{ paymentHold.holdId }}">
+                    <button class="govuk-button govuk-button--secondary" data-module="govuk-button" style="margin-bottom: 0px;">Remove</button>
```
Comment (steve-dickinson) [inline app/views/payment-holds.njk:52]: Clicking on `Remove` doesn't remove the hold

Comment (steve-dickinson) [inline app/views/payment-holds.njk:52]: Clicking on `Remove` doesn't remove the `hold`.

The `hold` in `ffc-pay-processing` has been closed though

File: app/routes/holds.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    },
+    handler: async (request, h) => {
+      await post('/add-payment-hold', { holdCategoryId: request.payload.holdCategoryId, frn: request.payload.frn }, null)
+      return h.redirect('/payment-holds')
+    }
+  }
+},
+{
+  method: 'POST',
```
Comment (steve-dickinson) [inline app/routes/holds.js:41]: Should this be `PUT`? Doesn't feel right having two `POST`

Comment (steve-dickinson) [inline app/routes/holds.js:41]: Should this be `PUT`? Doesn't feel right having two `POST`

File: app/views/add-payment-hold.njk [hunk 6]
```diff
@@ -23,47 +23,50 @@
...
-                  <span class="govuk-visually-hidden">Error:</span> {{ errorDetail.message }}
-                </span>
-              {% endfor %}
-            </ul>
+            <div class="govuk-error-summary__body">
+              <ul class="govuk-list govuk-error-summary__list">
+                {% for errorDetail in errors.details %}
+                  <span class="govuk-error-message">
+                    <span class="govuk-visually-hidden">Error:</span>
+                    {{ errorDetail.message }}
```
Comment (steve-dickinson) [inline app/views/add-payment-hold.njk:35]: The error which is returned for hold category missing is `"holdCategoryId" is required` would this be better as `"Hold category" is required`?


The values aren't retain if there is an error on another element eg. If FRN is correct, but a hold category is not selected, an error is returned, but the FRN is lost.

Comment (steve-dickinson) [inline app/views/add-payment-hold.njk:35]: The error which is returned for hold category missing is `"holdCategoryId" is required` would this be better as `"Hold category" is required`?

File: app/routes/home.js (hunk 1 above)
Comment (johnwatson484) [reply app/routes/home.js]: Yes it should.  Fixed.

File: app/views/home.njk (hunk 2 above)
Comment (johnwatson484) [reply app/views/home.njk:23]: Fixed as part of above

Comment (johnwatson484) [reply app/views/home.njk:23]: Fixed as part of above

File: app/views/payment-schemes.njk (hunk 3 above)
Comment (johnwatson484) [reply app/views/payment-schemes.njk:40]: Yes, when there are multiple secondary buttons are best and should have a load of primary buttons as it defeats the point of them being primary.  Took inspiration from feedback given to the Siti Agri development.  

The actual GDS advice is to not have a flood of any button but as these are "doing" something for the layout we have it feels the best.  We can validate all this with our users though and iterate it to fit their feedback

Comment (johnwatson484) [reply app/views/payment-schemes.njk:40]: As above

File: app/views/payment-holds.njk (hunk 4 above)
Comment (johnwatson484) [reply app/views/payment-holds.njk:52]: Fixed.  I wasn't filtering out closed holds as I wrongly assumed the API I'm calling only returned open holds.  Maybe it should longer term or those payloads are going to be huge in time.

Comment (johnwatson484) [reply app/views/payment-holds.njk:52]: Thinking about it, I'm definitely going to create a ticket or performance is going to suffer in a years time.

File: app/routes/holds.js (hunk 5 above)
Comment (johnwatson484) [reply app/routes/holds.js:42]: Can't do PUT from HTML.

Comment (johnwatson484) [reply app/routes/holds.js:42]: Can't do PUT from html

File: app/views/add-payment-hold.njk (hunk 6 above)
Comment (johnwatson484) [reply app/views/add-payment-hold.njk:35]: Updated the error message for both properties

Comment (johnwatson484) [reply app/views/add-payment-hold.njk:35]: Retained FRN on validation fail

Comment (johnwatson484) [reply app/views/add-payment-hold.njk:35]: Updated as above

**Pull request merged**